### PR TITLE
Restrict name chars

### PR DESCRIPTION
**Pull request by chouseknecht**
Allow only [a-zA-Z0-9-_] in Namespace.name and Repository.name.

### CODE REVIEW

File: galaxy/api/views/repository.py [hunk 1]
```diff
@@ -51,6 +52,16 @@ def get_repo(provider_namespace, user, repo_name):
     return repo
 
 
+def check_name(name):
+    if not name:
+        raise ValidationError(detail={'name': 'Value required'})
+
+    if re.search('[^A-Za-z0-9-_]', name):
```
Comment (cutwater) [inline galaxy/api/views/repository.py]: IMHO `re.match('^[\w-]+$')` is slightly better for readability, but this code is totally fine.

File: galaxy/api/views/namespace.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 # along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
 
 import logging
+import re
```
[the remark this answers is not in the document]
Comment (chouseknecht) [reply galaxy/api/views/namespace.py]: Good idea. However, the code is already doing validation in the view. We can refactor it later, if necessary.

**Comment by chouseknecht** [comment]
@cutwater 

Changed the regex to use `^[\w-]+$`, as you suggested. Much cleaner. Thanks!

It now allows A-Z in namespace, and performs the duplicate check using `iexact`. Regarding index performance, please review https://code.djangoproject.com/ticket/3575, and see if we need to change how we're indexing Namespace.

File: galaxy/api/views/repository.py [hunk 3]
```diff
@@ -51,6 +52,16 @@ def get_repo(provider_namespace, user, repo_name):
     return repo
 
 
+def check_name(name):
```
[the remark this answers is not in the document]
Comment (chouseknecht) [reply galaxy/api/views/repository.py]: Don't disagree. In the interest of time, added comments. We definitely need to add unit testing in a future release.

**Pull request merged**